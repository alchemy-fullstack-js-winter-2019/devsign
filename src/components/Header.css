a:link {
  text-decoration: none;
}

nav {
  border: 2px solid white;
  width: 80vw;
  text-align: right;
  margin: auto;

  a {
    display: inline-block;
    margin: 0 2vw;
    color: rgb(121, 8, 121);
    font-weight: bold;
  }
}