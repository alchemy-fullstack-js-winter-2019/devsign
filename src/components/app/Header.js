import React from 'react';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import ModalContainerChirp from '../../containers/ModalContainerChirp';
import styled from 'styled-components';

const Section = styled.section`
.warble {
  font-size: 4em;
  margin: 10px auto 0 auto;
  font-weight: 300;
  text-align: center;
  }
  .nav {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-top: 0;
    padding: 5px 15px 0px 15px;
    border-top: 2px solid black;
    border-bottom: 2px solid black;
  }
  .home {
    font-size: 2em;
    color: black;
  }
  input {
    display: flex;
    position: relative;
    margin: 10px auto;
  }
`;

export default function Header() {
  return (
    <Section>
      <header>
        <h1 className="warble">Warble</h1>
        <nav className="nav">
          <Link to="/home">
            <FontAwesomeIcon icon="home" className="home"/>
          </Link>
          <ModalContainerChirp />
          <Link to="/profile"><img src="../assets/prof-pic.png" width="40px" alt="prof pic logo"/></Link>
        </nav>
        <input type="text" placeholder="Search" name="search" />
      </header>
    </Section>
  );
}
  
  



