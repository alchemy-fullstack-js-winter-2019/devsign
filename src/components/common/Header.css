nav {
  display: flex;
  justify-content: flex-end;

  a {
    margin-right: 1em;
  }
}

header {
  margin-top: 1em;
  display: flex;
  flex-direction: row;
  align-items: center;
}

img {
  border-radius: 100%;
  height: 7em;
  width: 8em;
}

a {
  margin: 0;
  text-align: center;
  display: inline;
	position: relative;
	font: 1em Helvetica, Sans-Serif;
	letter-spacing: 2px;
  color: rgba(0,0,255,1);
  margin-left: -.5em;
}

.logo {
  animation: myLogo 5s infinite;
}

@keyframes myLogo {
  from {transform: rotateY(0deg)}
  to {transform: rotateY(360deg)}
}