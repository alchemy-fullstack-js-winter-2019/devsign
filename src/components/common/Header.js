import React from 'react';
import {
  Link
} from 'react-router-dom';
import styles from './Header.css';




export default function Header() {
  return (
    <header>
      <img src="src/assets/Robin.png" alt='rockin robin- twitter clone logo' className={styles.logo}/>
      <nav>
        <Link to='/'> Home </ Link>
        <Link to='/profile'> Profile </ Link>
        <Link to='/login'> Login </ Link>
        <Link to='/about'> About </ Link>
      </nav>
    </header>
  );
}
