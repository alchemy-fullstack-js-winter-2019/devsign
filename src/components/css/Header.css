  .header{
  background-color: black;
  color: white;
  margin: auto;

  h1 {
    display: flex;
    justify-content: center;
  }
  
  ul {
    display: flex;
    justify-content: flex-end;
    list-style-type: none;
    align-items: center;
    padding: 20px;
  }

  a {
    color: white;
  }

  li {
    padding: 10px;
  }
}

