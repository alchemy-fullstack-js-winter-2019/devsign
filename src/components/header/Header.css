*{
  font-family: 'Raleway', sans-serif;
}

header {
  h1 {
    font-size: 1.75em;
    color: #DEEBED;
    text-align: center;
    text-shadow: 1px 1px #D8D8D8;
  }

  nav{
    background-color: #D8D8D8;
    padding: 1px;
    text-align: center;
    a {
      padding: 5px;
      color: white;
      font-size: 1em;
      text-decoration: none;
    }
    :hover {
      color: black;
    }
  }
  
  
}
