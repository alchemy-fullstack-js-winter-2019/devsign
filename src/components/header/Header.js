import React from 'react';
import styles from './Header.css';

import logoPic from '../../../assets/tornado.svg';


export default function Header() {
  return (
    <header className={styles.Header}>
      <img src={logoPic}/>
      <h1>TWISTLER</h1>
      <h2>the only weather twitter app</h2>
    </header>
  );
};
