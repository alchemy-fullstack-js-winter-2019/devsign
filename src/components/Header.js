import React from 'react';
import { Link } from 'react-router-dom';
import './Header.css';

export default function Header() {
  return (
    <header>
      <nav>
        <Link to="/toptweets">Top Tweets</Link>
        <Link to="/register">SignUp/In</Link>
        <Link to="/"><img alt={'user profile picture'} src="https://cdn.shopify.com/s/files/1/1061/1924/products/Thinking_Face_Emoji_large.png?v=1480481060" /></Link>
      </nav>
    </header>
  );
}
