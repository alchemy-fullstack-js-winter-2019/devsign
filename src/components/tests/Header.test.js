import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../app/Header';
import { Provider } from 'react-redux';
import store from '../../store';
import { MemoryRouter } from 'react-router-dom';
import { library } from '@fortawesome/fontawesome-svg-core';
import { faHome, faWindowClose } from '@fortawesome/free-solid-svg-icons';

library.add(faHome, faWindowClose);

jest.mock('../../services/auth.js');

describe('Header', () => {
  it('matches a snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
