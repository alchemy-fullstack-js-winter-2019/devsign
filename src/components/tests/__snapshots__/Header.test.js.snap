// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot can match a snapshot 1`] = `
Array [
  <p>
    My Blog
  </p>,
  <nav
    className="Header"
  >
    <span>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </span>
    <span>
      <a
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </span>
    <span>
      <a
        href="/posts"
        onClick={[Function]}
      >
        Posts
      </a>
    </span>
    <span>
      <a
        href="/about"
        onClick={[Function]}
      >
        About
      </a>
    </span>
  </nav>,
]
`;
